src/main/java/org/generationcp/middleware/api/location/LocationServiceImpl.java

```diff
@@ -61,6 +64,27 @@ public long countFilteredLocations(final LocationSearchRequest locationSearchReq
 	@Override
 	public void deleteLocation(final Integer locationId) {
 		final Location location = this.daoFactory.getLocationDAO().getById(locationId);
+		List<Locdes> Blocks = null;
+		if (location.getLtype() == 415) { // Field
+			//Get the Blocks
+			Blocks = daoFactory.getLocDesDao().getLocdes(null, locationId.toString());
+			Blocks.forEach((locdes) -> {
+					this.daoFactory.getLocDesDao().makeTransient(locdes);
+					final Location blockLocation = this.daoFactory.getLocationDAO().getById(locdes.getLocationId());
+					this.daoFactory.getLocationDAO().makeTransient(blockLocation);
+				}
+			);
+
+			final List<Locdes> fields = daoFactory.getLocDesDao().getLocdes(locationId, null);
+			fields.forEach(locdes -> this.daoFactory.getLocDesDao().makeTransient(locdes));
+		}
+		if (location.getLtype() == 416) { // Block
+			Blocks = daoFactory.getLocDesDao().getLocdes(locationId, null);
+			Blocks.forEach(locdes -> {
```

Comment:
@cuenyad not clear to me why in this case we save locdes?

Reply:
done.